Swift/Sources/StateMachineMacros/Macros/StateMachineHashableMacro.swift

```
@@ -0,0 +1,93 @@
+
+        var associatedValueCases: [String] = []
+        for element: EnumCaseElementSyntax in elements {
+            if let parameters: EnumCaseParameterListSyntax = element.parameterClause?.parameters, !parameters.isEmpty {
+                if parameters.count > 1 {
+                    let associatedValues: String = (1...parameters.count)
+                        .map { "value\($0)" }
+                        .joined(separator: ", ")
+                    let `case`: String = """
+                        case let .\(element.name.text)(\(associatedValues)):
+                        return (\(associatedValues))
+                        """
+                    associatedValueCases.append(`case`)
+                } else {
+                    let `case`: String = """
+                        case let .\(element.name.text)(value):
+                        return (value)
+                        """
+                    associatedValueCases.append(`case`)
+                }
+            } else {
+                let `case`: String = """
+                    case .\(element.name.text):
+                    return ()
+                    """
+                associatedValueCases.append(`case`)
+            }
+        }
+
+        let decl: DeclSyntax = """
+            extension \(type): StateMachineHashable {
+
+                enum HashableIdentifier {
+
+                    \(raw: enumCases.joined(separator: "\n"))
+                }
+
+                var hashableIdentifier: HashableIdentifier {
+                    switch self {
+                    \(raw: hashableIdentifierCases.joined(separator: "\n"))
+                    }
+                }
+
+                var associatedValue: Any {
+                    switch self {
+                    \(raw: associatedValueCases.joined(separator: "\n"))
+                    }
+                }
+            }
+            """
+
+        return decl.as(ExtensionDeclSyntax.self).flatMap { [$0] } ?? []
```

Might consider throwing an informational error here rather than return an empty syntax, although I understand that an error here is probably not a client developer error, but an unexpected internal issue with this macro that the client developer may not be able to do something about specifically.

---

Done 👍